microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+    are included in initrd, and are all the same version.
+    """,
+)
+class HvModule(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the list of given LIS kernel modules and verify if the version
+           matches with the Linux kernel release number. (Drivers loaded directly in
+           to the kernel are skipped)
+        """,
+        priority=0,
+    )
+    def verify_lis_modules_version(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        if not isinstance(node.os, Redhat):
+            log.info(f"{node.os.name} not supported.")
+            raise SkippedException("This test case only supports Redhat distros.")
+
+        lis_driver = node.tools[LisDriver]
+        lis_version = lis_driver.get_version()
+        hv_modules = self.__get_modules_not_in_kernel(node)
+        lis_installed = node.os.package_exists("microsoft-hyper-v")
+
+        if lis_installed:
+            modinfo = node.tools[Modinfo]
+            for module in hv_modules:
+                module_version = VersionInfo.parse(modinfo.get_version(module))
+                assert_that(module_version).described_as(
+                    f"Version of {module} does not match LIS version"
+                ).is_equal_to(lis_version)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Verify the presence of all Hyper V drivers using lsmod
+           to look for the drivers not directly loaded into the kernel.
+        """,
+        priority=0,
+    )
+    def verify_hyperv_modules(
+        self, case_name: str, log: Logger, node: RemoteNode
+    ) -> None:
+        hv_modules = self.__get_modules_not_in_kernel(node)
+        distro_version = node.os.information.version
+
+        # Some versions of RHEL and CentOS have the LIS package installed
+        #   which includes extra drivers
+        if isinstance(node.os, Redhat):
+            modprobe = node.tools[Modprobe]
+            lis_installed = node.os.package_exists("microsoft-hyper-v")
+
+            if lis_installed:
+                hv_modules.append("pci_hyperv")
+                modprobe.run("pci_hyperv", sudo=True)
+
+            if (
+                distro_version >= "7.3.0" or distro_version < "7.5.0"
+            ) and lis_installed:
+                hv_modules.append("mlx4_en")
+                modprobe.run("mlx4_en", sudo=True)
+
+        # Counts the Hyper V drivers loaded as modules
+        num_modules_loaded = 0
+        lsmod = node.tools[Lsmod]
+        for module in hv_modules:
+            if lsmod.module_exists(module):
+                log.info(f"Module {module} present")
+                num_modules_loaded = num_modules_loaded + 1
+            else:
+                log.error(f"Module {module} absent")
+
+        num_modules_expected = len(hv_modules)
+        assert_that(num_modules_loaded).described_as(
```

Comment:
Check the length of list may be better, it may print the list, if the size is not expected.